qualtran/bloqs/qsp/polynomial_approximations.py

```diff
@@ -0,0 +1,87 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import bisect
+
+import numpy as np
+import scipy
+
+
+def degree_jacobi_anger_approximation(t: float, *, precision: float) -> int:
```

Comment:
Let's maybe modify the return type so that we return the symbolic expression encoded in `HamiltonianSimulationByGQSP.degree` if any of `t` or `precision` is symbolic here? Then the hamiltonian simulation class can always delegate to `degree_jacobi_anger_approximation` to find the degree and the symbolic expression can be used outside the context of hamiltonian simulation as well. 

There is nothing special about the symbolic expression that makes it specific to the Hamiltonian simulation case; correct?

Reply:
yep, does not depend on the use case. moved it now.